libsolidity/codegen/CompilerUtils.cpp

```diff
@@ -1014,6 +1014,8 @@ void CompilerUtils::convertType(
 			// stack: <address> <function_id>
 			m_context << Instruction::POP;
 		}
+		else if (stackTypeCategory == Type::Category::Function && targetTypeCategory == Type::Category::Function)
+			solAssert(_typeOnStack.isImplicitlyConvertibleTo(_targetType), "Invalid function type conversion requested.");
```

Comment:
This looks a little dangerous to me. Please add asserts that we only allow internal -> internal or external -> external conversions - there are so many different function types.

Reply:
Oh and please also add an assert that the stack size is equal.

Reply:
And what about the code that would have been generated otherwise in line 1023 - is the condition always false in this case?

Reply:
Updated.